node/tasks.py

```diff
@@ -219,37 +222,28 @@ def send_dispute_requests() -> None:
             "signature": utils.eth_sign(f'{zconfig.SEQUENCER["id"]}{timestamp}'),
         }
     )
+    apps_missed_txs:dict[str, Any] = {}
 
-    for node in zconfig.NODES.values():
-        if node["id"] in {zconfig.NODE["id"], zconfig.SEQUENCER["id"]}:
-            continue
-
-        for app_name in zconfig.APPS.keys():
-            missed_txs: dict[str, Any] = zdb.get_missed_txs(app_name)
-            if len(missed_txs) == 0:
-                continue
-
-            try:
-                response: dict[str, Any] | None = send_dispute_request(
-                    node, app_name, [tx["body"] for tx in missed_txs.values()]
-                )
-                if not response:
-                    continue
-
+    check_missed_txs = False
+    for app_name in zconfig.APPS.keys():
+        app_missed_txs = zdb.get_missed_txs(app_name)
+        apps_missed_txs[app_name] = app_missed_txs
+        if not check_missed_txs and len(app_missed_txs) != 0:
+            check_missed_txs = True
+    
+    if not check_missed_txs and not zdb.is_sequencer_down:
+        return
+    
+    tasks = [send_dispute_request(node, apps_missed_txs, zdb.is_sequencer_down) 
+            for node in zconfig.NODES.values()]
+    try:
+        completed = await asyncio.wait_for(asyncio.gather(*tasks), timeout=timeout)
```

Comment:
This should not wait for all but should check if dispute is approved one by one and add stake of approver one by one and proceed with  switch as soon as threshold has met.

Reply:
Fixed in this [commit](https://github.com/zellular-xyz/zsequencer/commit/197472cecd9bfbee894b35387d3bcc3504692b74).